File: server/makc2022--nodejs--nestjs/src/layer3/nosql-mongo/sample/mappers/typegoose/entities/dummy-one-to-many/mapper-dummy-one-to-many-entity.extension.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { DummyOneToManyEntityLoader } from 'src/layer3/nosql-mongo/sample/entities/dummy-one-to-many/dummy-one-to-many-entity.loader';
import { DummyOneToManyEntityObject } from 'src/layer3/nosql-mongo/sample/entities/dummy-one-to-many/dummy-one-to-many-entity.object';
import { MapperDummyOneToManyEntityObject } from './mapper-dummy-one-to-many-entity.object';

/** Расширение сущности "DummyOneToMany" сопоставителя. */
export class MapperDummyOneToManyEntityExtension {
  private constructor(
    private self: {
      entityObject?: DummyOneToManyEntityObject;
      mapperObject?: MapperDummyOneToManyEntityObject;
    }
  ) {}

  /** Расширить объект сущности.
   * @param entityObject Объект сущности.
   */
  static extEntityObject(entityObject: DummyOneToManyEntityObject) {
    return new MapperDummyOneToManyEntityExtension({ entityObject });
  }

  /** Расширить объект сопоставителя.
   * @param mapperObject Объект сопоставителя.
   */
  static extMapperObject(mapperObject: MapperDummyOneToManyEntityObject) {
    return new MapperDummyOneToManyEntityExtension({ mapperObject });
  }

  /** Преобразовать из объекта сущности в объект сопоставителя.
   * @returns Объект сопоставителя.
   */
  fromEntityToMapperObject(): MapperDummyOneToManyEntityObject {
    const result = new MapperDummyOneToManyEntityObject();

    new DummyOneToManyEntityLoader(result).load(this.self.entityObject);

    return result;
  }

  /** Преобразовать из объекта сопоставителя в объект сущности.
   * @returns Объект сущности.
   */
  fromMapperToEntityObject(): DummyOneToManyEntityObject {
    const loader = new DummyOneToManyEntityLoader();

    loader.load(this.self.mapperObject);

    return loader.entityObject;
  }
}

File: server/makc2022--dotnet/Makc2022.Layer3.Sql.Sample.Mappers.EF/Entities/DummyOneToMany/MapperDummyOneToManyEntityExtension.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2022.Layer3.Sql.Sample.Entities.DummyOneToMany;

namespace Makc2022.Layer3.Sql.Sample.Mappers.EF.Entities.DummyOneToMany
{
    /// <summary>
    /// Расширение сущности "DummyOneToMany" сопоставителя.
    /// </summary>
    public static class MapperDummyOneToManyEntityExtension
    {
        #region Public methods

        /// <summary>
        /// Преобразовать из объекта сущности в объект сопоставителя.
        /// </summary>
        /// <param name="entityObject">Объект сущности.</param>
        /// <returns>Объект сопоставителя.</returns>
        public static MapperDummyOneToManyEntityObject FromEntityToMapperObject(
            this DummyOneToManyEntityObject entityObject
            )
        {
            MapperDummyOneToManyEntityObject result = new();

            new DummyOneToManyEntityLoader(result).Load(entityObject);

            return result;
        }

        /// <summary>
        /// Преобразовать из объекта сопоставителя в объект сущности.
        /// </summary>
        /// <param name="mapperObject">Объект сопоставителя.</param>
        /// <returns>Объект сущности.</returns>
        public static DummyOneToManyEntityObject FromMapperToEntityObject(
            this MapperDummyOneToManyEntityObject mapperObject
            )
        {
            DummyOneToManyEntityLoader loader = new();

            loader.Load(mapperObject);

            return loader.EntityObject;
        }

        #endregion Public methods
    }
}
